packages/ui-core/src/components/Search/Search.test.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+import Search , { ISearchProps } from './Search';
+import cnCreate from 'utils/cnCreate';
+
+let mockedDebounce;
+jest.mock('lodash.debounce', () => {
+    mockedDebounce = jest.fn().mockImplementation((fn) => fn);
+    return mockedDebounce;
+});
+
+const cn = cnCreate('mfui-search');
+
+const props: ISearchProps = {
+    value: 'initial value',
+    placeholder: 'type to search here',
+    items: ['title', 'title2', 'title3', 'title4'],
+};
+
+describe('<Search />', () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('snapshots', () => {
+        it('renders Search without props correctly', () => {
+            const wrapper = shallow(<Search />);
+            expect(wrapper).toMatchSnapshot();
+        });
+
+        it('renders Search with all props correctly', () => {
+            const wrapper = shallow(<Search {...props} />);
+            expect(wrapper).toMatchSnapshot();
+        });
+
+        it('renders Search without icon', () => {
+            const wrapper = shallow(<Search {...props} hideIcon />);
+            expect(wrapper).toMatchSnapshot();
+        });
+    });
+
+    describe('typing', () => {
+        it('highlighted text', () => {
+            const wrapper = shallow(<Search {...props} />);
+
+            wrapper.find(`.${cn('search-field')}`).simulate('change', {
+                target: {
+                    value: 'title2',
+                },
+            });
+
+            expect(wrapper).toMatchSnapshot();
+        });
+        it('calls onChange while typing in the field with delay', () => {
+            jest.useFakeTimers();
+
+            const handleChange = jest.fn();
+            const wrapper = shallow(<Search {...props} changeDelay={300} onChange={handleChange} />);
+
+            wrapper.find(`.${cn('search-field')}`).simulate('change', {
+                target: {
+                    value: 'new value',
+                },
+            });
+
+            jest.advanceTimersByTime(300);
```

Comment:
Сейчас тест не проверяет, что handleChange не был вызван до задержки.
```suggestion
            jest.advanceTimersByTime(250);
            expect(handleChange).notHaveBeenCalled(1);
            jest.advanceTimersByTime(300);
            expect(handleChange).toHaveBeenCalledWith('new value');
```